controllers/csm_controller.go

```diff
@@ -615,96 +673,107 @@ func (r *ContainerStorageModuleReconciler) getDriverConfig(ctx context.Context,
+		log.Errorw("error delete node service account", "Error", err.Error())
 		return err
 	}
 
-	replicationEnabled, clusterClients, err := utils.GetDefaultClusters(ctx, instance, r)
-	if err != nil {
+	if err = utils.DeleteObject(ctx, &driverConfig.Controller.Rbac.ServiceAccount, cluster.ClusterCTRLClient); err != nil {
+		log.Errorw("error delete controller service account", "Error", err.Error())
 		return err
 	}
-	for _, cluster := range clusterClients {
-		if err = utils.DeleteObject(ctx, &driverConfig.Node.Rbac.ServiceAccount, cluster.ClusterCTRLClient); err != nil {
-			log.Errorw("error delete node service account", "Error", err.Error())
-			return err
-		}
 
-		if err = utils.DeleteObject(ctx, &driverConfig.Controller.Rbac.ServiceAccount, cluster.ClusterCTRLClient); err != nil {
-			log.Errorw("error delete controller service account", "Error", err.Error())
-			return err
-		}
+	if err = utils.DeleteObject(ctx, &driverConfig.Node.Rbac.ClusterRole, cluster.ClusterCTRLClient); err != nil {
+		log.Errorw("error delete node cluster role", "Error", err.Error())
+		return err
+	}
 
-		if err = utils.DeleteObject(ctx, &driverConfig.Node.Rbac.ClusterRole, cluster.ClusterCTRLClient); err != nil {
-			log.Errorw("error delete node cluster role", "Error", err.Error())
-			return err
-		}
+	if err = utils.DeleteObject(ctx, &driverConfig.Controller.Rbac.ClusterRole, cluster.ClusterCTRLClient); err != nil {
+		log.Errorw("error delete controller cluster role", "Error", err.Error())
+		return err
+	}
 
-		if err = utils.DeleteObject(ctx, &driverConfig.Controller.Rbac.ClusterRole, cluster.ClusterCTRLClient); err != nil {
-			log.Errorw("error delete controller cluster role", "Error", err.Error())
-			return err
-		}
+	if err = utils.DeleteObject(ctx, &driverConfig.Node.Rbac.ClusterRoleBinding, cluster.ClusterCTRLClient); err != nil {
+		log.Errorw("error delete node cluster role binding", "Error", err.Error())
+		return err
+	}
 
-		if err = utils.DeleteObject(ctx, &driverConfig.Node.Rbac.ClusterRoleBinding, cluster.ClusterCTRLClient); err != nil {
-			log.Errorw("error delete node cluster role binding", "Error", err.Error())
-			return err
-		}
+	if err = utils.DeleteObject(ctx, &driverConfig.Controller.Rbac.ClusterRoleBinding, cluster.ClusterCTRLClient); err != nil {
+		log.Errorw("error delete controller cluster role binding", "Error", err.Error())
+		return err
+	}
 
-		if err = utils.DeleteObject(ctx, &driverConfig.Controller.Rbac.ClusterRoleBinding, cluster.ClusterCTRLClient); err != nil {
-			log.Errorw("error delete controller cluster role binding", "Error", err.Error())
-			return err
-		}
+	if err = utils.DeleteObject(ctx, driverConfig.ConfigMap, cluster.ClusterCTRLClient); err != nil {
+		log.Errorw("error delete configmap", "Error", err.Error())
+		return err
+	}
+
+	if err = utils.DeleteObject(ctx, driverConfig.Driver, cluster.ClusterCTRLClient); err != nil {
```

Comment:
Is this guaranteed that we delete the target driver only if the Operator initially created it? Could this cause any issue if we want to keep the driver installed on the target cluster?